src/test/java/math/EloRatingSnippetTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2017-2019 Ilkka Seppälä
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package math;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+/*
+ * Tests for 30 Seconds of Java code library
+ *
+ */
+class EloRatingSnippetTest {
+  /**
+   * Tests for EloRatingSnippet. {@link EloRatingSnippet#eloAdjustment(int,int,double,double)}
+   */
+  @Test
+  void testEloRating() {
+    assertEquals(1400, EloRatingSnippet.eloAdjustment(1400, 1400, 0.5, 32));
+    assertEquals(1416, EloRatingSnippet.eloAdjustment(1400, 1400, 1.0, 32));
+    assertEquals(1384, EloRatingSnippet.eloAdjustment(1400, 1400, 0.0, 32));
+    assertEquals(1885, EloRatingSnippet.eloAdjustment(1900, 1400, 0.5, 32));
+    assertEquals(1901, EloRatingSnippet.eloAdjustment(1900, 1400, 1.0, 32));
+    assertEquals(1869, EloRatingSnippet.eloAdjustment(1900, 1400, 0.0, 32));
+  }
```

Comment:
Looks good. Where do you get the values for the tests?

Reply:
Any tests for the `expectedWinProbability`?

Reply:
The values are arbitrary, but also come from personal experience e.g. two equally rated players drawing should net 0 rating adjustment. The rest was calculating and validating. 

Yes, I've added tests for `expectedWinProbability`, thank you for that mention. I also went ahead with adding rounding to the logic. This hopefully clarifies the results a bit, but can be reverted.

Reply:
Sorry, K-Factor values are also arbitrary (here it is given as 32). My understanding is this is partly why they classify Elo as a rating system rather than an exact formula. Organizations have adopted their own process for finding the K-Factor, but it ranges from about 10 to 40. 

For example, it looks like the Internet Chess Club (ICC) uses a static K=32, whereas the US Chess Federation uses different values depending on rating: K=32 for those below 2100, K=24 for those between 2100 and 2400, and K=16 for those above 2400. This just means stronger, more tenured players are subject to less swing in their ratings, and the opposite is true for new players.